src/accel/RunAction.cc

```diff
@@ -59,12 +83,122 @@ void RunAction::EndOfRunAction(const G4Run*) {}
 void RunAction::build_core_params()
 {
     G4AutoLock lock(&mutex);
```

Comment:
Should the mutex be a static local variable? I'm sure geant4 would put it at global scope but seems to me like we shouldn't have to...

Reply:
Don't see why not.